tools/sotn_build/get_file_list.py

```diff
@@ -0,0 +1,311 @@
+        and assetFilters == (("", ""),)
+    ):
+        assetFilters = tuple(
+            tuple(f.split("*"))
+            for f in ("D_801*.bin", "*.gfxbin", "*.palbin", "cutscene_*.bin")
+        )
+
+    # The file listing is really the only one used currently because some asset segment files are built by splat extensions
+    # and the resulting files aren't directly specified in the config
+    if os.path.isdir(assetPath):
+        assetFiles = {
+            os.path.join(assetPath, file)
+            for file in os.listdir(assetPath)
+            if any(
+                file.startswith(sFilter) and file.endswith(eFilter)
+                for sFilter, eFilter in assetFilters
+            )
+        }
+    else:  # This method is included as something of a proof of concept/work in progress
+        # Splat configs have different segment type names than what these files are saved as, so we translate them with this dict
+        # dict.get() can have a default value specified, so we use extension.get(seg_type,seg_type) to get the translation
+        # If there is no translation in the dict, then it just uses seg_type
+        extension = {
+            "cmp": "dec",
+            "raw": "bin",
+            "ci4": "png",
+            "palette": "png",
+            "assets": "json",
+        }
+        assetFiles = {
+            FILENAME_TEMPLATE.substitute(
+                path=assetPath, filename=seg[2], ext=extension.get(seg[1], seg[1])
+            )
+            for seg in segments
+            if len(seg) >= 3 and seg[1] in SEG_TYPES["asset"]
+        }
+        # Add a .bin file for each .dec file and supplement assetFiles with a directory listing
+        assetFiles.update(
+            {file.replace("dec", "bin") for file in assetFiles if ".dec" in file}
+        )
+    return assetFiles
+
+
+# Dedicated function for ParseArgs so that this file can be imported in a module and still have the ability to parse out the proper args
+def ParseArgs(args: list):
+    parser = argparse.ArgumentParser(
+        description="For gathering information in more complex ways than Make is meant to do"
+    )
+    parser.add_argument(
+        "--version",
+        required=False,
+        type=str,
+        default=os.getenv("VERSION") or "us",
+        help="Sets game version and overrides VERSION environment variable",
+    )
+    subparsers = parser.add_subparsers(dest="action", help="The action to perform")
+    parserList = subparsers.add_parser("list", help="Retrieve a list of files")
+    parserList.add_argument("ovl", type=str, help="Overlay to target")
+    parserList.add_argument(
+        "list_type",
+        type=str,
+        choices=("src_files", "o_files", "merged"),
+        help="What to list",
+    )
+    parserList.add_argument(
+        "-c",
+        "--config-type",
+        type=str,
+        choices=list(CONFIG_PATTERNS.keys()),
+        required=True,
+        help="Type of config to reference",
```

Comment:
Am I misunderstanding, or is this only `splat` currently?
What would be the future use of this parameter?